metricbeat/module/postgresql/replication/data.go

```diff
@@ -0,0 +1,53 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package replication
+
+import (
+	"time"
+
+	s "github.com/elastic/beats/v7/libbeat/common/schema"
+	c "github.com/elastic/beats/v7/libbeat/common/schema/mapstrstr"
+)
+
+// Based on https://www.postgresql.org/docs/9.2/static/monitoring-stats.html#PG-STAT-DATABASE-VIEW
+var schema = s.Schema{
+	"pid": c.Int("pid"),
+	"user": s.Object{
+		"id":   c.Int("usesysid", s.Optional),
+		"name": c.Str("usename"),
+	},
+	"application_name": c.Str("application_name"),
+	"client": s.Object{
+		"address":  c.Str("client_addr"),
+		"hostname": c.Str("client_hostname"),
+		"port":     c.Int("client_port", s.Optional),
+	},
+	"backend_start": c.Time(time.RFC3339Nano, "backend_start"),
+	"backend_xmin":  c.Time(time.RFC3339Nano, "backend_xmin", s.Optional),
```

Comment:
Are you sure RFC3339Nano should be used? Ref?

Reply:
Ref:
- https://www.postgresql.org/docs/current/datatype-datetime.html
- https://www.postgresql.org/docs/current/datatype-datetime.html#:~:text=tomorrow%27%3A%3Adate.-,8.5.2.%C2%A0Date/Time%20Output,-The%20output%20format

Reply:
I was just confirming. You changed it to RFC3339 which is not correct IMO. Postgres has precision up to 1 microsecond (if not nanosecond). RFC3339 Nano is fine is the datetime format is RFC3339. The recent change you made is not correct.